modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostGISDialect.java

```diff
@@ -687,6 +688,56 @@ public void handleUserDefinedType(
         }
     }
 
+    /** Using PostGIS table spatial_ref_sys to determine authority name and code of srid */
+    @Override
+    public CoordinateReferenceSystem createCRS(int srid, Connection cx) throws SQLException {
+        if (srid <= 0) {
+            return null;
+        }
+        try (Statement statement = cx.createStatement()) {
+            String sqlStatement =
+                    "SELECT AUTH_NAME, AUTH_SRID, SRTEXT FROM SPATIAL_REF_SYS WHERE SRID = "
+                            + Integer.toString(srid);
```

Comment:
Unnecessary toString, that's what string concatenation does already. Also, please create the sqlStatement outside of the try, and then you can create both the statement and the resultSet in the same try-with-resources block, reducing nesting.